File: backend/app/api/v1/irrigation.py
from fastapi import APIRouter
from app.schemas.irrigation import IrrigationRequest
from app.schemas.common import APIResponse
from app.services.irrigation_service import IrrigationService

router = APIRouter()

@router.post("/recommend", response_model=APIResponse)
async def recommend(data: IrrigationRequest):
    result = await IrrigationService.recommend(data)
    return APIResponse(data=result)

File: frontend/src/api/irrigation.ts
import apiClient from './client';
import type { IrrigationRequest, IrrigationResponse } from '../types/irrigation';

export const getRecommendation = async (data: IrrigationRequest): Promise<IrrigationResponse> => {
  const response = await apiClient.post('/irrigation/recommend', data);
  return response.data;
};
